src/libmongoc/tests/TestSuite.c

```diff
@@ -906,26 +882,48 @@ TestSuite_TestMatchesName (const TestSuite *suite,
 }
 
 
+bool
+test_matches (TestSuite *suite, Test *test)
+{
+   int i;
+   bool matches;
+
+   /* If no match patterns were provided, then assume all match. */
+   if (suite->match_patterns.len == 0) {
+      matches = true;
+   } else {
```

Comment:
nit: What you have here is fine, but the function is a bit simpler when the positive if-case just returns and this "else" is the rest of the function.

Reply:
Updated and removed the `matches` variable. An earlier iteration of these changes supported a `--skip` and `--after` flag. I removed them because they were unnecessary, but did not clean up this function.

> @kevinAlbs wiki.c2.com/?ElseConsideredSmelly is a good read if you have some spare time

Oh thank you, I have added that to my reading list.